TypeCobol/Compiler/Scanner/TokensGroup.cs

```diff
@@ -77,5 +77,30 @@ public override string ToString()
             }
             return sb.ToString();
         }
+
+        /// <summary>
+        /// Text returned to the parser :
+        /// - Token.SourceText if the token is not a PartialCobolWord
+        /// - The concatenation of ToString() of all OriginalTokens.
+        /// </summary>
+        public override string SourceText
+        {
+            get
+            {
+                if (this.TokenType != TokenType.PartialCobolWord)
+                    return base.SourceText;
+                else 
+                {
+                    StringBuilder sb = new StringBuilder();
+                    foreach (Token token in OriginalTokens)
+                    {
+                        sb.Append(token.SourceText);
+                    }
+                    String t = sb.ToString();
+                    return t;
```

Comment:
Maybe this should be generalized for all TokensGroup not only PartialCobolWord ? I thinl it is more related to the fact that original tokens span over multiple lines instead of TokenType equals to PartialCobolWord.

Reply:
done in fe1bab6